podpac/core/coordinates/uniform_coordinates1d.py

```diff
@@ -384,12 +384,23 @@ def copy(self):
 
     def _select(self, bounds, return_indices, outer):
         # TODO is there an easier way to do this with the new outer flag?
+        my_bounds = self.bounds.copy()
 
-        lo = max(bounds[0], self.bounds[0])
-        hi = min(bounds[1], self.bounds[1])
+        # If the bounds are of instance datetime64, then the comparison should happen at the lowest precision
+        if self.name == "time" and isinstance(my_bounds[0], np.datetime64) and isinstance(bounds[0], np.datetime64):
+            mine = np.timedelta64(1, my_bounds[0].dtype.name.replace("datetime64", "")[1:-1])
+            your = np.timedelta64(1, bounds[0].dtype.name.replace("datetime64", "")[1:-1])
 
-        fmin = (lo - self.bounds[0]) / np.abs(self.step)
-        fmax = (hi - self.bounds[0]) / np.abs(self.step)
+            if mine > your:
+                bounds = [b.astype(my_bounds[0].dtype) for b in bounds]
+            else:
+                my_bounds = [b.astype(bounds[0].dtype) for b in my_bounds]
```

Comment:
Can you help me understand why this is necessary? This is called by the `select` method you modified in the parent class, so the bounds should already be changed be the time we get here.

If not, factor this out and use the utility method I proposed above.

Reply:
The `bounds` get changed, that's fine, but `my_bounds` do not, so we need to do it again to `my_bounds`. We don't want to change `my_bounds` in-place in the object because other parts of the pipeline might still need the higher precision.

Reply:
Ah, right, okay. Well, let's factor that out then, if we can.